metadata-ingestion/setup.py

```diff
@@ -512,7 +512,7 @@
     "flake8-tidy-imports>=4.3.0",
     "flake8-bugbear==23.3.12",
     "isort>=5.7.0",
-    "mypy==1.0.0",
+    "mypy",
```

Comment:
let's keep it pinned e.g. `mypy==1.10.1`. mypy new versions have caused CI breakages before, which I'd like to avoid